midx.c

```diff
@@ -1040,15 +1062,32 @@ int verify_midx_file(const char *object_dir)
 	}
 
 	progress = start_progress(_("Verifying object offsets"), m->num_objects);
+
+	/*
+	 * Create an array mapping each object to its packfile id.  Sort it
+	 * to group the objects by packfile.  Use this ordering to visit each
+	 * of the objects, but only require 1 packfile to be open at a time.
+	 */
+	ALLOC_ARRAY(pairs, m->num_objects);
 	for (i = 0; i < m->num_objects; i++) {
+		pairs[i].pos = i;
+		pairs[i].pack_int_id = nth_midxed_pack_int_id(m, i);
+	}
+	QSORT(pairs, m->num_objects, compare_pair_pos_vs_id);
+
+	for (k = 0; k < m->num_objects; k++) {
 		struct object_id oid;
 		struct pack_entry e;
 		off_t m_offset, p_offset;
 
-		nth_midxed_object_oid(&oid, m, i);
+		if (k > 0 && pairs[k-1].pack_int_id != pairs[k].pack_int_id &&
+		    m->packs[pairs[k-1].pack_int_id])
+			close_pack_fd(m->packs[pairs[k-1].pack_int_id]);
```

Comment:
If I'm reading this correctly, this is an (optional) optimization that will keep the pack files open to a minimum.  I'm assuming without it, they would start being closed transparently as you reached some max threshold.  Since you know they are sorted, makes sense to do the optimization here.

Reply:
yeah, the problem with max file descriptors that i fixed yesterday said we were holding 2000+ packfiles open when we started running out.  fixing that caused us to still hold 2000+ open, but close and open packfiles as necessary to do the random access.

So yeah, this fix kinda eliminates the need for the previous fix.  But i'm keeping that one in for now since it is harmless and just seems like the correct thing to do.